controllers/gitrepository_controller.go

```diff
@@ -440,9 +442,17 @@ func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 		conditions.Delete(obj, sourcev1.SourceVerifiedCondition)
 	}
 
-	// Configure authentication strategy to access the source
-	var authOpts *git.AuthOptions
-	var err error
+	u, err := url.Parse(obj.Spec.URL)
+	if err != nil {
+		e := serror.NewStalling(
+			fmt.Errorf("failed to parse url '%s': %w", obj.Spec.URL, err),
+			sourcev1.URLInvalidReason,
+		)
+		conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
+		return sreconcile.ResultEmpty, e
+	}
```

Comment:
We should move this closer to where `u` is being dereferenced, as we have no validation there which makes it harder to understand whether that operation is `safe`.